### PR TITLE
Feature/tech items select

### PR DESCRIPTION
**Pull request by JoshuaHinman**
# Description

This PR adds a new endpoint `voyages/{teamId}/techs/selections`, allowing teams to designate tech stack items as "Selected" within their categories.  It enforces a maximum of 3 selections per category, defined by `const MAX_SELECTION_COUNT`.

The teamTechStackItem model has been updated with Boolean value `isSelected`, so it is important to update with `yarn migrate`  , and `yarn push`

There are e2e test for the following responses:
-200 successful update
-400 max selections per category exceeded
-400 invalid teamid
-401 unauthorized- *uses old JWT test*

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Feature updates / changes
- [ ] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

tested with swagger and `yarn test:e2e techs`

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

File: src/techs/techs.service.ts
Comment (timDeHof) [inline src/techs/techs.service.ts:117]: I checkout the branch locally and ran `yarn start:dev` that gave me an error saying "error TS2353: Object literal may only specify known properties, and 'isSelected' does not exist in type '(Without<TeamTechStackItemUpdateInput, TeamTechStackItemUncheckedUpdateInput> & TeamTechStackItemUncheckedUpdateInput) | (Without<...> & TeamTechStackItemUpdateInput)'."
on line 117 techs.service.ts. 

was there any changes done to the prisma schema?

File: test/techs.e2e-spec.ts
Comment (timDeHof) [inline test/techs.e2e-spec.ts:56]:
```suggestion
    describe("GET voyages/teams/:teamId/techs - get data on all tech categories and items", () => {
```

File: test/techs.e2e-spec.ts
Comment (timDeHof) [inline test/techs.e2e-spec.ts:112]:
```suggestion
    describe("POST voyages/teams/:teamId/techs - add new tech item", () => {
```

File: src/techs/techs.service.ts
Comment (JoshuaHinman) [reply src/techs/techs.service.ts:117]: Yes - it was the first change I made and I forgot to mention it!  `yarn migrate` is needed to update

File: test/techs.e2e-spec.ts
Comment (timDeHof) [inline test/techs.e2e-spec.ts:217]:
```suggestion
    describe("POST voyages/teams/:teamId/techs/:teamTechId - add user vote for tech item", () => {
```

File: test/techs.e2e-spec.ts
Comment (timDeHof) [inline test/techs.e2e-spec.ts:310]:
```suggestion
    describe("DELETE voyages/teams/:teamId/techs/:teamTechId - delete user vote for tech", () => {
```

File: test/techs.e2e-spec.ts
Comment (timDeHof) [inline test/techs.e2e-spec.ts:403]:
```suggestion
    describe("PATCH voyages/teams/:teamId/techs/selections - updates isSelected value of tech stack items", () => {
```

File: test/techs.e2e-spec.ts
Comment (timDeHof) [inline test/techs.e2e-spec.ts:408]:
```suggestion
                .patch(`/voyages/teams/${teamId}/techs/selections`)
```

File: test/techs.e2e-spec.ts
Comment (timDeHof) [inline test/techs.e2e-spec.ts:442]:
```suggestion
                .patch(`/voyages/teams/${teamId}/techs/selections`)
```

File: test/techs.e2e-spec.ts
Comment (timDeHof) [inline test/techs.e2e-spec.ts:498]:
```suggestion
                .patch(`/voyages/teams/${teamId}/techs/selections`)
```

File: test/techs.e2e-spec.ts
Comment (JoshuaHinman) [reply test/techs.e2e-spec.ts:112]: Thanks - I was just going to change these

**Review by timDeHof — CHANGES_REQUESTED** [review]
After running `yarn migrate:test:docker`, I'm still getting the same errors about `isSelected` not  existing in type. Also it looks like the prisma schema with the updated models were pushed to the remote branch.

**Comment by timDeHof** [comment]
I'm sorry to do this to you. But you might want to wait on changing the endpoint paths until my PR is merge. otherwise, your test would fail with your latest changes.

**Comment by siasktv** [comment]
Hi Josh! the code looks good to me. I just got this error on the branch after `yarn migrate`   
image

**Comment by cherylli** [comment]
Might need to pull dev into this branch, it's still showing /voayges/{teamId}/techs on swagger and controller for me, so all tests failed (404 not found)

File: src/techs/techs.controller.ts
Comment (cherylli) [inline src/techs/techs.controller.ts:191]:
```suggestion
            "Maximum of 3 selections per category allowed. Login required. All tech items (isSelected === true/false) are required",
```

**Review by cherylli — COMMENTED** [review]
It works fine, if the request body is accurate. I did something like if there's already 3 trues in the same categorie in the database, you could add another 2 true with something like this 

```
{
  "categories": [
    {
      "categoryId": 1,
      "techs": [
        {
          "techId": 9,
          "isSelected": true
        },
{
          "techId": 12,
          "isSelected": true
        }
      ]
    }
  ]
}
```

as shown 5 items are selected now

image

File: src/techs/techs.controller.ts
Comment (JoshuaHinman) [reply src/techs/techs.controller.ts:191]: That is a good catch.  Yes, it is assumed that all techs in a category will be submitted together in the same array.  It counts all `true` values from this array, and does not check the DB.  Also - the `categoryId` field is not used and is only there for readability at this point.

**Comment by timDeHof** [comment]
Great job! All tests passed and it worked on swagger.approved.

**Review by timDeHof — APPROVED** [review]

**Comment by JoshuaHinman** [comment]
This branch had several issues, so I pulled a new one from dev branch and re applied the new code

**Pull request closed without merging**
